src/subapps/search/containers/SearchContainer.tsx

```diff
@@ -87,40 +93,80 @@ const SearchContainer: React.FC = () => {
     });
   }
 
-  const { columns, data } = useGlobalSearchData(
+  const {
+    columns,
+    data,
+    updateFieldsVisibility,
+    updateAllColumnsToVisible,
+    fieldsVisibility,
+    setFieldsVisibility,
+    visibleColumns,
+  } = useGlobalSearchData(
     query,
     pagination.currentPage,
     pagination.pageSize,
     onQuerySuccess,
     nexus
   );
 
+  function onUpdateColumnVisibilityFromPageSize(columnCount: number) {
+    const columnVisibilities = columns?.map((el, ix) => {
+      return {
+        name: el.label,
+        key: el.key,
+        visible: ix < columnCount,
+      };
+    });
+    columnVisibilities && setFieldsVisibility(columnVisibilities);
+  }
+
+  const { tableRef } = useColumnsToFitPage(
+    wrapperDOMProps,
+    onUpdateColumnVisibilityFromPageSize
+  );
+
   return (
     <TableHeightWrapper
       wrapperHeightRef={wrapperHeightRef}
       resultTableHeightTestRef={resultTableHeightTestRef}
       wrapperDOMProps={wrapperDOMProps}
     >
-      {columns && data && (
-        <Table
-          columns={columns}
-          dataSource={data}
-          pagination={{
-            showTotal,
-            onShowSizeChange,
-            total: pagination.totalNumberOfResults,
-            pageSize: pagination.pageSize,
-            current: pagination.currentPage,
-            onChange: handlePaginationChange,
-            position: ['topRight'],
-            locale: { items_per_page: '' },
-            showSizeChanger: true,
-            pageSizeOptions: pagination.pageSizeOptions,
-            showLessItems: true,
-          }}
-          rowKey="@id"
-          onRow={onRowClick}
-        ></Table>
+      {visibleColumns && data && (
+        <>
+          <div className="searchTableHeader">
```

Comment:
can we use hyphen-delimited syntax for class names instead of camel case please? for consistency